src/main/java/com/palonskiy/serice/BookServiceImpl.java

```diff
@@ -53,79 +47,41 @@ public void update(BookDto bookDto) {
 
     @Override
     public List<AuthorDto> getBookAuthors(int id) {
-        List<BookAuthor> bookAuthors = bookDao.getBookAuthors().stream()
-                .filter(bookAuthor -> (bookAuthor.getBook() == bookDao.getById(Long.valueOf(id))))
-                .collect(Collectors.toList());
-        List<Author> authors = new ArrayList<>();
-        for (BookAuthor bookAuthor : bookAuthors) {
-            authors.add(bookAuthor.getAuthor());
-        }
-        return AuthorConverter.toDtoList(authors);
+        return AuthorConverter.toDtoList(bookDao.getBookAuthors(Long.valueOf(id)));
     }
 
     @Override
     public void add(BookAuthorDto bookAuthorDto) {
-        if (bookAuthorDto.getBook() != null) {
-            if (bookAuthorDto.getAuthor() != null) {
-                if (bookDao.checkIfExist(bookAuthorDto.getBook().getName()) == null) {
-                    bookDao.add(BookConverter.toBook(bookAuthorDto.getBook()));
-                    Long bookId = bookDao.getId();
-                    Long authorId =bookAuthorDto.getAuthor().getId();
-                    if (authorDao.checkIfExist(bookAuthorDto.getAuthor().getSecondName()) == null) {
-                        authorDao.add(AuthorConverter.toAuthor(bookAuthorDto.getAuthor()));
-                        authorId = authorDao.getId();
-                    }
-                    bookAuthorService.add(bookAuthorDto, authorId, bookId);
-                } else throw new ExistingEntityException();
-            } else throw new NullBookException();
-        } else
-            throw new NullAuthorException();
-    }
-
-    @Override
-    public void addOnlyBook(BookDto bookDto) {
-        bookDao.add(BookConverter.toBook(bookDto));
-    }
-
-    @Override
-    public Long getId() {
-        return bookDao.getId();
+        if (bookAuthorDto.getBook() == null) {
+            throw new NullBookException();
+        }
+        if (bookAuthorDto.getAuthor() == null) {
+            new NullAuthorException();
+        }
+        if (!bookDao.checkIfExist(bookAuthorDto.getBook().getName())) {
+            Book book = bookDao.add(BookConverter.toBook(bookAuthorDto.getBook()));
+            Author author = authorService.add(bookAuthorDto.getAuthor());
+            book.setAuthors(Arrays.asList(author));
+            bookDao.update(book);
+        } else throw new ExistingEntityException();
     }
 
     @Override
     public void delete(int id) {
-        bookDao.delete(Long.valueOf(id));
+        bookDao.delete(bookDao.getById(Long.valueOf(id)));
```

Comment:
use either int or long for IDs. please, dont mix them

Reply:
Have changed dao and service layers to long IDs and add cast to long in controllers